internal/registry/registry.go

```diff
@@ -126,6 +126,30 @@ func (r Registry) searchImport(name string) (*Package, bool) {
 	return nil, false
 }
 
+// resolveImportConflict generates and assigns a unique alias for
+// packages with conflicting qualifiers.
+func (r Registry) resolveImportConflict(a, b *Package, lvl int) {
```

Comment:
While reviewing this code I wondered, if there are simpler ways to get a unique import alias. Two ideas I would like to share:

1. Instead of adding more levels of the import path to the alias only when needed, always add all the levels, because as mentioned in the comment of `uniqueName`, the full path is guaranteed to be unique.
1. Generate a hash from the full import path and appending some characters (similar to the shortened git hashes) from the hash to the original package name resulting in something like `pkg_a3b295`.

I think, with both approaches, the process for generating the aliases is straight forward without the need for any recursion and comparing.

In the end the code that is generated by `moq` is not intended to be as readable as possible by humans, where this code is written by humans and for humans.

Reply:
The current way of identifying an appropriate alias has had bugs and needed multiple patches. This clearly indicates that the problem is a bit difficult to fully understand and apply to all the different scenarios in practice. So there would definitely be justification for generating aliases that are guaranteed to not conflict.

However, I think one of the strengths of `moq` is that it generates code that looks hand-written with the benefit of readable and meaningful names with helpful comments. To that effect, I think there is value in bearing the cost of maintenance to a reasonable degree. This change is close to that limit though. So if more issues arise, would strongly consider moving in the direction suggested by you.

Hope that made sense.